### PR TITLE
Adds Wizard-Tracking crate to cargo

### PR DESCRIPTION
**Pull request by Robustin**
Lich cancer is metastasizing and locker wizards are as common as ever.

For 12,000 cargo points, YOU can help put an end to this nightmare with 3 pinpointers that will track the wizard. 

On an broader note, this will give the crew something tangible to work towards for wizard rounds instead of just sitting and waiting for the wizard to come murder their department. If the wizard wants to prevent it, they can raid cargo, and if the crew wants to prevent that, they can defend cargo. It gives the crew a place where they can finally force a confrontation with the lord of lockers.

### CODE REVIEW

**Comment by Bawhoppen** [comment]
wow this is retarded and makes no ic-sense

**Comment by Robustin** [comment]
Because wizards on a space station are going to produce rounds that are just full of ic-reasonability. 

Or everyone is running around turning each other into barnyard animals while a fucking lich-rat is ventcrawling around and mindswapping everyone into rats.

**Comment by coiax** [comment]
Five pinpointers for 10k? Pretty cheap, all things considered.

Maybe just make a regular pinpointer orderable for 10k, and make it easy for admins to add the wizard as a valid target. OR have liches valid pinpointer targets, because evil radiation energy and shit.

**Comment by Robustin** [comment]
Its already super easy for me to add wizards to standard pinpointers.

5 for 10 ain't a bad deal, but the issue is getting to 10 in the first place, especially when its not going to be a priority round-start kind of order. This doesn't give you magic wizard-killing powers, it just lets you track them in a way that isn't borderline impossible if there's no/dead AI.

**Comment by coiax** [comment]
Question, will this be introduced to catch those locker hiding/ventcrawling heads in rev?

**Comment by Bobylein** [comment]
+coiax heads can ventcrawl? o.O

File: code/game/gamemodes/nuclear/pinpointer.dm [hunk 1]
```diff
@@ -295,3 +295,12 @@
 			user << "Nearest operative detected is <i>[nearest_op.real_name].</i>"
 		else
 			user << "No operatives detected within scanning range."
+			
+/obj/item/weapon/pinpointer/wizard
+	name = "wiz pointer"
+	desc = "Tracks wizards with extreme precision; down the exact locker they are hiding in."
+		
+/obj/item/weapon/pinpointer/attack_self()
+	for(var/datum/mind/wizard in ticker.mode.wizards)
```
Comment (KorPhaeron) [inline code/game/gamemodes/nuclear/pinpointer.dm]: Can you actually do this? Does this work?

Comment (KorPhaeron) [inline code/game/gamemodes/nuclear/pinpointer.dm]: I dont think you're supposed to track an abstract concept

Comment (Robustin) [inline code/game/gamemodes/nuclear/pinpointer.dm]: Going to test it in a few hours and find out! If this doesn't work I think get_turf for the mind should.

Comment (KorPhaeron) [inline code/game/gamemodes/nuclear/pinpointer.dm]: if(wizard.current)
the_disk = wizard.current
break

**Comment by Shadowlight213** [comment]
If we're going to add wizard trackers, then you should remove the existing tells that lich has pointing to the vessel since you can just pinpointer to them.

File: code/game/gamemodes/nuclear/pinpointer.dm [hunk 2]
```diff
@@ -295,3 +295,12 @@
 			user << "Nearest operative detected is <i>[nearest_op.real_name].</i>"
 		else
 			user << "No operatives detected within scanning range."
+			
+/obj/item/weapon/pinpointer/wizard
+	name = "wiz pointer"
```
Comment (Iamgoofball) [inline code/game/gamemodes/nuclear/pinpointer.dm]: so does it point at the bathrooms?

File: code/modules/cargo/packs.dm [hunk 3]
```diff
@@ -118,6 +118,17 @@
 	crate_name = "special ops crate"
 	crate_type = /obj/structure/closet/crate
 
+/datum/supply_pack/emergency/wiztrackers
+	name = "Wizard Trackers"
+	cost = 10000
```
Comment (Iamgoofball) [inline code/modules/cargo/packs.dm]: buff it to 15K credits and drop the contents to 3 pinpointers

**Comment by AdamElTablawy** [comment]
>adding a specific anti-antag crate to cargo

how about you just remove lich

**Comment by Shadowlight213** [comment]
The reasoning behind it isn't lich specific though.
I haven't seen locker wizards be a huge epidemic tbh.

File: code/game/gamemodes/nuclear/pinpointer.dm (hunk 2 above)
Comment (Robustin) [inline code/game/gamemodes/nuclear/pinpointer.dm]: That effect is so broken it probably wouldn't make a difference, but no, the vast majority of wizard rounds I don't expect these trackers to come into play, lich still needs that downside in the likely event that trackers are not purchased.

**Comment by Robustin** [comment]
It's even more than just liches and lockers though. It comes back to jaunt/teleport letting wizards 100% disengage from almost any fight without a realistic chance of being pursued by the crew. Even with "Scyar Nila Arrivals!" you know by the time you get there he's jaunted to abandoned surgery, a toilet stall, the ian shirt room, etc. to laugh at everyone. 

These trackers don't give wizard hunters some kind of magic wiz-killing power, the ability to flee will still be robust, and just as importantly it's really not hard to disrupt cargo's operation as a wizard.

**Comment by Supermichael777** [comment]
soloution hide in cargo locker and ambush crate

**Comment by Incoming5643** [comment]
I have a nerf lined up for liches that is both agreeable and hilarious. This isn't the way.

Also it completely ruins the rare soul that still attempts stealth wiz.

**Comment by KorPhaeron** [comment]
Yeah, what happens if I mindswap with this?

**Comment by coiax** [comment]
It tracks the mind, Kor.

**Comment by TheNightingale** [comment]
This isn't cool for those times when you go mindswap and four robeless apprentices.

**Comment by KorPhaeron** [comment]
That's what I mean Coiax, seems to ruin mindswap

**Comment by TechnoAlchemisto** [comment]
Add antag buster crates for every antag to cargo please

**Comment by kilkun** [comment]
it's potentially good, but make it 1 pinpointer for 10k and if that doesn't solve it, increment it.

**Comment by Incoming5643** [comment]
There's nothing good about this

**Comment by kilkun** [comment]
anything that stops bad wizards from memeing in lockers is good by my standard.

**Comment by Incoming5643** [comment]
There's literally no reason to ever nerf wizards beyond when you want to nerf one spell for being too good compared to the rest and always being taken.

There is never, EVER, a good reason to suggest an across the board nerf.

**Comment by kilkun** [comment]
this isn't a nerf for anything but a shitty strategy used by terrible players who shouldn't even have wizard enabled. For every other wizard out there who is actually interested in not being a piece of shit meme who hides all round because they don't want to die, this changes pretty much nothing.

**Comment by Robustin** [comment]
Except its a way to deal with wizards that ISN'T a nerf. If you think the ideal balance for wizard is "the crew has no fucking clue where he is, no fucking clue where he's going, and aside from brief ~10 seconds bursts where he pops into someones office and fireballs, lightning + lasereyes, or mm + einaths them before disappearing again.

This isn't about one spell or even one tactic, it's about trying to make wizard into a mode that isn't scraping the bottom. Now that we're finally running out of meme items to give the wizard, people are confronting the ugly realities of wizard mode and I give it a year or two at most before it ends up in 0-2.5% weighting territory. 

Either some solution has to appear on the tracking horizon or we can remove blink/jaunt/teleport/lich and make the wizard something that could feasibly be tracked via conventional means. If the wizard doesn't want to get tracked he can teleport over to cargo and permanently disrupt their operations in a matter of seconds. It also gives security and the crew a tangible goal to shoot for in progress against the wizardl

I'll raise the cost and reduce the number of trackers, but its sad how many people want to defend the status quo on one of the worst modes. I'm sure plenty of you were arguing for the status quo of Malf up until Kor rightfully put it in the grave.

**Comment by Incoming5643** [comment]
Wizard is wonderful and I won't let anyone cock it up over a problem that doesn't exist on a code level.

Shitty wizards is a player problem, not a code problem.

**Comment by PKPenguin321** [comment]
>Except its a way to deal with wizards that ISN'T a nerf. If you think the ideal balance for wizard is "the crew has no fucking clue where he is, no fucking clue where he's going, and aside from brief ~10 seconds bursts where he pops into someones office and fireballs, lightning + lasereyes, or mm + einaths them before disappearing again.

If you wanna get rid of wizard whack-a-mole then make the wizard ultra-strong so he doesn't have to retreat every five seconds
I say give him natural immunity to explosions because *magic* or something, maybe give him a weak passive HP regen (it's magic), give the robes very high armor

**Comment by TechnoAlchemisto** [comment]
>webclient compile errors

love it

**Comment by KorPhaeron** [comment]
Here is a wizard nerf I would accept: Some sort of magic "school" system that prevents you from stacking up on 3 aoe stuns and 2 escapes. Make you pick from different things. That or a tree, but I realize that's way more complicated.

**Comment by KorPhaeron** [comment]
And yeah wizard play whack-a-mole because they're glass cannons, often with long cooldowns on their attacks

**Comment by optimumtact** [comment]
admins should be banning any wizards hiding in a locker

honestly they should be banning anyone who picks lich spell as well

**Comment by Ausops** [comment]
its alright admins will just go stealthmode and not answer ahelps when goof uses hellfoam on a wiz (and misses)

**Comment by coiax** [comment]
I still don't agree with this "you can do bannable shit to kill antags, but only if you succeed" policy. Tis like punishing attempted murder, but only if they DIDN'T manage to kill Hitler.

**Comment by Iamgoofball** [comment]
@Ausops absolutey

**Comment by Robustin** [comment]
Incoming you're crazy if you think Wizards are in good shape. I observe from an "impartial" position during most wizard rounds now and I have yet to see a wizard round that didn't ding our server pop. and produce a low quality round. Lich, Shapeshift, etc. are all the latest memes trying to keep Wizard on life-support but realistically Wizard either needs a bunch of changes like this PR to make the two sides interact in a more meaningful fashion, or someone has to completely overhaul wizard mechanics.

**Comment by PKPenguin321** [comment]
Yeah honestly we need more quick and painless wiz rounds, shit like lich makes it really drag out

At the same time though I'd like to make the wizard more durable to cheesier tactics like bomb spears though

**Comment by Incoming5643** [comment]
wizard rounds are already some of the quickest in the game

**Comment by Robustin** [comment]
They're only quick when a wizard cucks himself with fireball, and this PR has nothing to do with those rounds. Shapeshift was probably the straw that broke the camel's back for me. Lockerhiding and escapespam was bad enough, but having the wizard spend 80% of the round in the fucking vent system was angering me and I wasn't even playing. The crew deserves a way to combat these tactics.

**Comment by Robustin** [comment]
Going to close it because of 2much opposition, because apparently its a better idea to just rebuild wizard from the ground up than let cargo spend all their points on something that helps them find the round's only* antagonist when he's hiding in the fucking pipes as a rat-lich.

**Comment by Incoming5643** [comment]
Protip: If you can tell a wizard is hiding in the vents, introduce plasma to the vents.

There IS a counterplay here

**Comment by Robustin** [comment]
@Incoming5643 

It's very easy for me to tell that when I'm observing/adminning. 

Absolutely not easy to do that when the mouse is using fireball (which it can cast as a mouse YAY!) or mindswap or ei nath to make sure there are *no witnesses*. 

Even then, nobody would ever flood plasma because of the risk of it leaking/flooding unintentionally.

Even THEN, the shapeshift meta always has the wizard taking lich so that if someone stomps your rat you aren't actually dead forever. Bonus: When a lich uses RISE! as a shapeshifted creature it doesn't show the glowing stream to the phylactery, not like it matters because the lich can easily relocate the phylactery and the "stream" only lasts a couple seconds.

**Comment by Iamgoofball** [comment]
@Incoming5643 and then you get banned for flooding the station with plasma if you miss the wizard

**Pull request closed without merging**